src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/strategy/AbstractStrategy.java

```diff
@@ -0,0 +1,157 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.strategy;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.CcdAuthenticator;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.CcdAuthenticatorFactory;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.model.Envelope;
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDataContent;
+import uk.gov.hmcts.reform.ccd.client.model.Event;
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
+
+/**
+ * Implementation of strategy.
+ * Any strategy is invoking steps as follows:
+ * <ul>
+ *     <li>jurisdiction authentication</li>
+ *     <li>start the event</li>
+ *     <li>create data content accordingly</li>
+ *     <li>submit the event</li>
+ * </ul>
+ * Any strategy will have to be implemented as an example:
+ * <pre>{@code
+ * @Component("strategy-name")
+ * class StrategyName extends AbstractStrategy {
+ *
+ *     StrategyName() {
+ *         // any extra autowiring needed
+ *     }
+ *
+ *     @Override
+ *     Object mapEnvelopeToCaseDataObject(Envelope envelope) {
+ *         return null; // implement this
+ *     }
+ *
+ *     @Override
+ *     String getEventTypeId() {
+ *         return ""
+ *     }
+ *
+ *     @Override
+ *     String getEventSummary() {
+ *         return ""
+ *     }
+ * }}</pre>
+ * <p/>
+ * No need to make any public access as everything will be injected in {@link StrategyContainer}:
+ * <pre>{@code
+ * @Resource(name = "strategy-name")
+ * private Strategy someNamedStrategy;}</pre>
+ * <p/>
+ * Then include each strategy in {@link StrategyContainer}
+ */
+abstract class AbstractStrategy implements Strategy {
+
```

Comment:
Naming this Strategy tells me how you have implemented it, not what it does. Wouldn't it be better, for the sake of clarity, to name the classes and also the containing package (currently named strategy) based on what they do, not on the implementation details?

By the way this looks to me more like a "template method" pattern than a strategy pattern.